soc: arm: st_stm32: Move STM32H7_DUAL_CORE

Desvauxm-st opened the pull request:
Move STM32H7_DUAL_CORE to soc/arm/st_stm32/stm32h7/Kconfig.defconfig.series. Set default to y.
Cleanup occurences where was set to y.

erwango commented on soc/arm/st_stm32/stm32h7/Kconfig.defconfig.series:
```diff
@@ -17,4 +17,9 @@ config ROM_START_OFFSET
 	default 0x400 if BOOTLOADER_MCUBOOT
 	default 0x0   if !BOOTLOADER_MCUBOOT
 
+config STM32H7_DUAL_CORE
+	bool "Dual Core"
+	default y
```
Please put it in `soc/arm/st_stm32/stm32h7/Kconfig.soc`

erwango commented on soc/arm/st_stm32/stm32h7/Kconfig.soc:
```diff
@@ -77,4 +77,8 @@ config SOC_STM32H7B3XXQ
 	select CPU_CORTEX_M7
 	select CPU_HAS_FPU_DOUBLE_PRECISION
 
+config STM32H7_DUAL_CORE
+	bool "Dual Core"
+	default y
+
```
It should be moved outside of choice structure.

Desvauxm-st replied:
ok

FRASTM reviewed (commented):
CI / merge conflicts to fix
Else LGTM

Desvauxm-st committed:
soc: arm: st_stm32: Move STM32H7_DUAL_CORE

Move STM32H7_DUAL_CORE to
soc/arm/st_stm32/stm32h7/Kconfig.soc
add select STM32H7_DUAL_CORE for SOC_STM32H745XX/H747XX
Cleanup old occurences where was set to y

Signed-off-by: Marc Desvaux <marc.desvaux-ext@st.com>

erwango commented on boards/arm/nucleo_h723zg/Kconfig.defconfig:
```diff
@@ -15,4 +15,7 @@ config NET_L2_ETHERNET
 
 endif # NETWORKING
 
+config STM32H7_DUAL_CORE
+	default n
```
Not required. Kconfig symbols are `n` by default

erwango commented on boards/arm/nucleo_h743zi/Kconfig.defconfig:
```diff
@@ -15,4 +15,7 @@ config NET_L2_ETHERNET
 
 endif # NETWORKING
 
+config STM32H7_DUAL_CORE
+	default n
```
Ditto, please remove

erwango commented on soc/arm/st_stm32/stm32h7/Kconfig.soc:
```diff
@@ -78,3 +78,8 @@ config SOC_STM32H7B3XXQ
 
 endchoice
+
+config STM32H7_DUAL_CORE
+	bool "Dual Core"
+	depends on SOC_SERIES_STM32H7X
+	default y
```
Don't default to `y`.
Use `select STM32H7_DUAL_CORE` in matching SoC definitins, like:
```
config SOC_STM32H747XX
	bool "STM32H747XX"
	select CPU_HAS_FPU_DOUBLE_PRECISION if CPU_CORTEX_M7
        select STM32H7_DUAL_CORE
```

(might require to move `STM32H7_DUAL_CORE`  definition before the choice section.

Desvauxm-st replied:
ok

erwango reviewed (changes requested):
Please update commit message which is not matching changesanymore.
Otherwise LGTM, thanks
